python/lsst/daf/butler/core/config.py

```diff
@@ -25,11 +25,14 @@
 
 __all__ = ("Config", "ConfigSubset")
 
+from dataclasses import dataclass
 import collections
 import copy
 import logging
 import pprint
 import os
+import pkg_resources
+import posixpath
```

Comment:
You should add a comment to explain why this is preferred over `os.path` and `pathlib`.